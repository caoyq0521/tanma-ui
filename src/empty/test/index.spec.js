import { mount } from '@vue/test-utils';
import TmEmpty from '../empty.vue';

const AXIOM = 'tanma-fe';
describe('tmEmpty', () => {
  it('render empty', () => {
    const wrapper = mount(TmEmpty);
    expect(wrapper).toMatchSnapshot();
  });

  it('type', async () => {
    const wrapper = mount(TmEmpty);

    expect(wrapper.find('.tm-empty--empty').exists()).toBe(true);

    await wrapper.setProps({ type: 'error' });
    expect(wrapper.find('.tm-empty--error').exists()).toBe(true);

    await wrapper.setProps({ type: 'build' });
    expect(wrapper.find('.tm-empty--build').exists()).toBe(true);
  });

  it('image', () => {
    const wrapper = mount(TmEmpty, {
      propsData: {
        image: AXIOM
      }
    });

    expect(wrapper.find('.tm-empty__image img').exists()).toBe(true);
  });

  it('image slot', () => {
    const wrapper = mount({
      template: `<tm-empty>
        <template #image>
          {{ AXIOM }}
        </template>
      </tm-empty>`,
      data() {
        return {
          AXIOM
        };
      }
    });

    expect(wrapper.find('.tm-empty__image').html()).toContain(AXIOM);
  });

  it('imageSize', async () => {
    const wrapper = mount(TmEmpty, {
      propsData: {
        imageSize: 300
      }
    });

    expect(wrapper.find('.tm-empty__image').attributes('style')).toContain('width: 300px');

    await wrapper.setProps({ imageSize: 400 });
    expect(wrapper.find('.tm-empty__image').attributes('style')).toContain('width: 400px');

  });

  it('description', () => {
    const wrapper = mount(TmEmpty, {
      propsData: {
        description: AXIOM
      }
    });

    expect(wrapper.find('.tm-empty__description').html()).toContain(AXIOM);
  });

  it('description slot', () => {
    const wrapper = mount({
      template: `<tm-empty>
        <template #description>
          {{ AXIOM }}
        </template>
      </tm-empty>`,
      data() {
        return {
          AXIOM
        };
      }
    });

    expect(wrapper.find('.tm-empty__description').html()).toContain(AXIOM);
  });
})
  