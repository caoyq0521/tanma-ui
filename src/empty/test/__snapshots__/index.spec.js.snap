// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmEmpty render empty 1`] = `
<div class="tm-empty tm-empty--empty">
  <div class="tm-empty__image" style="width: 260px;"><img src="test-file-stub" alt=""></div>
  <div class="tm-empty__description">
    暂无数据~
  </div>
</div>
`;
